doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,428 @@
+  :cmd:`Section` may remove notations.  Some notations are defined in
+  the standard library (see :ref:`thecoqlibrary`) and are generally
+  loaded automatically at startup time.
+
+  Here are the character sequences that Coq directly defines as tokens
+  without using :cmd:`Notation`::
+
+    ! #[ % & ' ( () (bfs) (dfs) ) * ** + , - ->
+    . .( .. ... / : ::= := :> :>> ; < <+ <- <:
+    <<: <= = => > >-> >= ? @ @{ [ [= ] _
+    `( `{ { {| | |- || }
+
+  When multiple tokens match the beginning of a sequence of
+  characters, the longest matching token is used.  Occasionally you
+  may need to insert spaces to separate tokens.  For example, if ``~``
+  and ``~~`` are both defined as tokens, the inputs ``~ ~`` and ``~~``
+  generate different tokens, whereas if `~~` is not defined, then the
+  two inputs are equivalent.
+
+Basic notions
+-------------
+
+Glossary
+~~~~~~~~
+
+.. glossary::
+
+   term
+
+     Terms are syntactically valid objects constructed from the
+     following grammar.  When they are :term:`well-typed`, they
+     represent mathematical objects.
+
+     .. insertprodn term field_def
+
+     .. prodn::
+        term ::= @term_product_or_fun
+        | @term_let
+        | @term_if
+        | @term_fix
+        | @term_cofix
+        | @term100
+        term100 ::= @term_cast
+        | @term10
+        term10 ::= @term_application
+        | @one_term
+        one_term ::= @ @qualid_annotated
+        | @term1
+        term1 ::= @term_projection
+        | @term0 % @scope
+        | @term0
+        term0 ::= @qualid_annotated
+        | @sort
+        | @primitive_notations
+        | @term_evar
+        | @term_match
+        | @term_record
+        | @term_generalizing
+        | @term_ltac
+        | ( @term )
+        qualid_annotated ::= @qualid {? @univ_annot }
+
+     .. note::
+
+        Many commands and tactics use :n:`@one_term` rather than
+        :n:`@term`.  The former need to be enclosed in parentheses
+        unless they're very simple, such as a single identifier.  This
+        avoids confusing a space-separated list of terms or
+        identifiers with a :n:`@term_application`.
+
+   type
+
+     Types are a sub-class of the syntactical class of terms:
+
+     .. prodn::
+        type ::= @term
+
+     Not every term is a type, but every term should have an
+     associated type, which is determined by applying a set of typing
+     rules.  If that's indeed the case, we say that the term is
+     :gdef:`well-typed`.
+
+     Intuitively, types may be viewed as sets containing terms.  We
+     say that a type is :gdef:`inhabited` if it contains at least one
+     term (i.e. if we can find a term which is associated to this
+     type).
+
+     Formally, this gives rise to logical foundations which are not
+     the same as the well-known `"set theory"
+     <https://en.wikipedia.org/wiki/Set_theory>`_.  Coq is based on
+     the Calculus of Inductive Construction which is a `"type theory"
```

Comment:
I thought it was "Constructions" not "Construction"?

Reply:
That's a typo, thanks.